### ISSUE TITLE
Cannot find package "github.com/atlassian/go-artifactory/v2/artifactory"

### ISSUE THREAD

**Issue by FrankTheTank9** [atlassian/terraform-provider-artifactory#33]
Hi,

We've been using the provider v1.2.3 since a long time with no issue but started getting the following error on Friday March 15th:

go get -d github.com/atlassian/terraform-provider-artifactory
package github.com/atlassian/go-artifactory/v2/artifactory: cannot find package "github.com/atlassian/go-artifactory/v2/artifactory" in any of:
	/usr/lib/go/src/github.com/atlassian/go-artifactory/v2/artifactory (from $GOROOT)
	/opt/go/src/github.com/atlassian/go-artifactory/v2/artifactory (from $GOPATH)

We saw this package was added as part of the v1.5 merge: https://github.com/atlassian/terraform-provider-artifactory/commit/94bde0db8e5bc257bb2375ed2950e72348640acc#diff-bdca0546c31d141072c28c7f5905df41.

We tried updating to 1.5 but got the same error.

Could you confirm if this is an issue with the provider itself? If not, any idea why we are getting this error?

Thanks!

**Comment by dillon-giacoppo** [comment]
Hi, to help debug this issue please provide clear steps to reproduce. The following are required, go version, the output of go get with '-v' used.

If possible please also try a clean install with a version of go > 0.11. Create a directory outside of your GOPATH: 

```bash
mkdir -p /tmp/tf-test && cd /tmp/tf-test
git clone https://github.com/atlassian/terraform-provider-artifactory.git
cd terraform-provider-artifactory
GO111MODULE=on go build
```

From the logs you provided, it looks like you may not be using [go modules](https://github.com/golang/go/wiki/Modules) (since it specifies the GOROOT) but is impossible to tell without further info.

**Comment by dillon-giacoppo** [comment]
**tl;dr was an issue with module backwards compatibility and go-artifactory HEAD being master and not V2. This has been fixed and should now work.**

I was able to replicate the error on go 1.11. 

```bash
docker run -it --rm --entrypoint bash golang:1.11
go get -v github.com/atlassian/terraform-provider-artifactory
```

This is a combination of not using module awareness and the default 
branch for go-artifactory being master. It seems the module aware compatibility 
layer was not able to work if the v2 module is not the HEAD branch.

The first workaround was:
```bash
# This will work without module support
cd $GOPATH/src/github.com/atlassian/go-artifactory/
git checkout v2
cd /
go get -v github.com/atlassian/terraform-provider-artifactory
```

It appears changing the default branch in github for go-artifactory to 
V2 is enough to make the original command work:
```bash
go get -v github.com/atlassian/terraform-provider-artifactory
```

I would personally not recommend installing the provider this way as it does not
pin to a version. The `go get` command has a variety of different behaviours 
that are project/directory/environment specific.

The ideal installation is (which should work with any version of golang >= 1.9)
```bash
git clone https://github.com/atlassian/terraform-provider-artifactory.git && cd terraform-provider-artifactory
git checkout v1.5.0 # Or a version you prefer
go install
```

Or alternatively (but less preferred due to the inconsistent nature of go get):
```bash
GO111MODULE=on go get -v github.com/atlassian/terraform-provider-artifactory@v1.5.0

# or if you prefer a floating version
GO111MODULE=on go get -v github.com/atlassian/terraform-provider-artifactory
```

Or you can use the provided docker image:
```bash
docker run -it --rm -v $(pwd):/data/src -w /data/src dillongiacoppo/terraform-artifactory:latest plan   
```

**Comment by FrankTheTank9** [comment]
Thanks for the quick fix and for the recommendation. I confirm this is now working for us.
